api/clever/__init__.py

```diff
@@ -224,59 +250,57 @@ def remote_patron_lookup(self, token):
             with the data listed above.
 
         """
-        bearer_headers = {
-            'Authorization': 'Bearer %s' % token
-        }
+        bearer_headers = {'Authorization': 'Bearer %s' % token}
         result = self._get(self.CLEVER_API_BASE_URL + '/me', bearer_headers)
         data = result.get('data', {}) or {}
 
         identifier = data.get('id', None)
 
         if not identifier:
-            return INVALID_CREDENTIALS.detailed(
-                _("A valid Clever login is required.")
-            )
+            return INVALID_CREDENTIALS.detailed(lgt("A valid Clever login is required."))
 
         if result.get('type') not in self.SUPPORTED_USER_TYPES:
             return UNSUPPORTED_CLEVER_USER_TYPE
 
         links = result['links']
 
-        user_link = [l for l in links if l['rel'] == 'canonical'][0]['uri']
+        user_link = [link for link in links if link['rel'] == 'canonical'][0]['uri']
+        # The canonical link includes the API version, so we use the base URL.
         user = self._get(self.CLEVER_API_BASE_URL + user_link, bearer_headers)
 
         user_data = user['data']
         school_id = user_data['school']
-        school = self._get(
-            self.CLEVER_API_BASE_URL + '/v1.1/schools/%s' % school_id,
-            bearer_headers
-        )
-
+        school = self._get(f"{self.CLEVER_API_BASE_URL}/v1.1/schools/{school_id}", bearer_headers)
         school_nces_id = school['data'].get('nces_id')
 
         # TODO: check student free and reduced lunch status as well
 
         if school_nces_id is None:
-            self.log.error(
-                "No NCES ID found in Clever school data: %s", repr(school)
-            )
+            self.log.error("No NCES ID found in Clever school data: %s", repr(school))
             return CLEVER_UNKNOWN_SCHOOL
 
         if school_nces_id not in TITLE_I_NCES_IDS:
             self.log.info("%s didn't match a Title I NCES ID", school_nces_id)
             return CLEVER_NOT_ELIGIBLE
 
+        external_type = None
+
```

Comment:
I don't think we need this, since `external_type` is set in all branches of the upcoming `if`.